scribus/langmgr.cpp

```diff
@@ -89,90 +89,545 @@ void LanguageManager::generateLangList()
-	m_langTable.append(LangDef("th",       "th_TH",    "Thai",                    QObject::tr("Thai")));
-	m_langTable.append(LangDef("tr",       "tr_TR",    "Turkish",                 QObject::tr("Turkish")));
-	m_langTable.append(LangDef("uk",       "uk_UA",    "Ukranian",                QObject::tr("Ukranian")));
-	m_langTable.append(LangDef("ur",       "",         "Urdu",                    QObject::tr("Urdu")));
-	m_langTable.append(LangDef("vi",       "",         "Vietnamese",              QObject::tr("Vietnamese")));
-	m_langTable.append(LangDef("cy",       "cy_GB",    "Welsh",                   QObject::tr("Welsh")));
+	m_langTable.append(LangDef("aiw",      "",         "Aari",                                QObject::tr("Aari")));
+	m_langTable.append(LangDef("abq",      "",         "Abaza",                               QObject::tr("Abaza")));
+	m_langTable.append(LangDef("ab",       "",         "Abkhazian",                           QObject::tr("Abkhazian")));
+	m_langTable.append(LangDef("ach",      "",         "Acoli",                               QObject::tr("Acoli")));
+	m_langTable.append(LangDef("ady",      "",         "Adyghe",                              QObject::tr("Adyghe")));
+	m_langTable.append(LangDef("prs",      "",         "Afghan Persian/Dari",                 QObject::tr("Afghan Persian/Dari")));
+	m_langTable.append(LangDef("af",       "af_ZA",    "Afrikaans",                           QObject::tr("Afrikaans")));
+	m_langTable.append(LangDef("aa",       "",         "Afar",                                QObject::tr("Afar")));
+	m_langTable.append(LangDef("aio",      "",         "Aiton",                               QObject::tr("Aiton")));
+	m_langTable.append(LangDef("ak",       "",         "Akan",                                QObject::tr("Akan")));
+	m_langTable.append(LangDef("sq",       "",         "Albanian",                            QObject::tr("Albanian")));
+	m_langTable.append(LangDef("atv",      "alt",      "Altai",                               QObject::tr("Altai")));
+	m_langTable.append(LangDef("am",       "",         "Amharic",                             QObject::tr("Amharic")));
+	m_langTable.append(LangDef("ar",       "",         "Arabic",                              QObject::tr("Arabic")));
+	m_langTable.append(LangDef("an",       "an_ES",    "Aragonese",                           QObject::tr("Aragonese")));
+	m_langTable.append(LangDef("hy",       "",         "Armenian",                            QObject::tr("Armenian")));
+	m_langTable.append(LangDef("rup",      "",         "Aromanian/Arumanian/Macedo-Romanian", QObject::tr("Aromanian/Arumanian/Macedo-Romanian")));
+	m_langTable.append(LangDef("frp",      "",         "Arpitan/Francoprovençal",             QObject::tr("Arpitan/Francoprovençal")));
+	m_langTable.append(LangDef("as",       "",         "Assamese",                            QObject::tr("Assamese")));
```

Comment:
Just choose one of the two names here, either “Arpitan” or “Francoprovençal”. This should happen for all languages with multiple names on the list, just pick the most common one (e.g. the Wikipedia article name for that language).